### PR TITLE
Improve doc for auto compaction

### PR DESCRIPTION
**Pull request by jihoonson**
I crosslinked pages for compaction configuration, coordinator API, and description of segment optimization. I also added a section for how to check the compaction is needed.

### CODE REVIEW

File: docs/content/design/coordinator.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ To ensure an even distribution of segments across Historical nodes in the cluste
 ### Compacting Segments
 
 Each run, the Druid Coordinator compacts small segments abutting each other. This is useful when you have a lot of small
-segments which may degrade the query performance as well as increasing the disk space usage.
+segments which may degrade the query performance as well as increasing the disk space usage. See [Segment Size Optimization](../operations/segment-optimization.html) for details.
```
Comment (justinborromeo) [inline docs/content/design/coordinator.md]: "segments which may degrade the query performance as well as increasing the disk space usage" -> "segments which may degrade query performance as well as increase disk space usage"

Comment (jihoonson) [reply docs/content/design/coordinator.md]: Thanks, fixed!

File: docs/content/operations/segment-optimization.md [hunk 2]
```diff
@@ -32,15 +32,57 @@ In Druid, it's important to optimize the segment size because
   which hold the input segments of the query. Each node has a processing threads pool and use one thread per segment to
   process it. If the segment size is too large, data might not be well distributed over the
   whole cluster, thereby decreasing the degree of parallelism. If the segment size is too small,
-  each processing thread processes too small data. This might reduce the processing speed of other queries as well as
-  the input query itself because the processing threads are shared for executing all queries.
+  each processing thread might process too small data. This can reduce the overall processing speed because
+  parallel processing involves some overhead like thread scheduling.
```
[the remark this answers is not in the document]
Comment (jihoonson) [reply docs/content/operations/segment-optimization.md]: Looks good to me. Thanks!

File: docs/content/operations/segment-optimization.md [hunk 3]
```diff
@@ -32,15 +32,57 @@ In Druid, it's important to optimize the segment size because
...
+  datasource = 'your_dataSource' AND
+  is_published = 1
+GROUP BY 1, 2, 3
+ORDER BY 1, 2, 3 DESC;
+```
+
+Please note that the query result might include overshadowed segments.
+In this case, you may want to see only rows of the max version per interval (pair of `start` and `end`).
+
+The recomended number of rows per segment and segment size are 5 million rows and 300 ~ 700MB, respectively.
```
[the remark this answers is not in the document]
Comment (jihoonson) [reply docs/content/operations/segment-optimization.md]: Sounds good. Moved this part and emphasized the importance of # of rows.

**Pull request merged**